app/models/user.rb

```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  validates :email, presence: true, format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: :create }
```

Comment:
3️⃣ since we are only collecting an email and password I think if they use a different email, it would register as a whole new account. I know that is the case for me. I have to make sure I sign in with the same email to get access to the data I created earlier (I have both a personal and teaching email that I use). 

4️⃣ I  choose to use Regex to validate the email because it should not need any maintenance. I read this thread and choose it as the best option: https://stackoverflow.com/questions/38611405/email-validation-in-ruby-on-rails
It is also listed as one of the Ruby/ Rails validators on this link: https://api.rubyonrails.org/v5.1/classes/ActiveModel/Validations/ClassMethods.html#method-i-validates
Do you want me to add these links as comments to the code?

Side note: we can definitely pass this through to the FE when we build the Post path. Right now, there is no create user API route as this is only the database creation. Agreed, when we create the route we should return the username.

Reply:
I think you make a good point that if we are going to do anything with updating email addresses, that is a discussion and separate issue down the road. Let's put some blinders on and not feature creep on this issue right now.

I think those are great resources to have used as a reference so I'm glad you added them to the comments for the PR for later reference. What I'm talking about for code comments is a comment detailing exactly what the format that the regex is looking for in case we need to change what a valid format looks like down the road or we have issues with the format validations. making it easier to debug, break down, and understand exactly what the regex code says since it is unfamiliar and very specific. (eg: `This checks that there is at least one character of any type non-case sensitive followed by an @ symbol ...`)